### PR TITLE
feat(route53resolver-alpha): fetch managed domain list id using custom resource

### LINKED ISSUE #25614: (module name): (short issue description)
### Describe the feature

Automatically pulling the ID for an AWS Managed Domain List for use with the aws-route53resolver (@aws-cdk/aws-route53resolver-alpha module). I know this is an alpha module but I thought I would get this feature at least since typically you deploy to multiple regions/accounts. The ID is unique in every region/account.

### Use Case

I want to deploy to multiple environments easily without having to do a hacky workaround like pulling the list of IDs using AWS SDK within my CDK Construct.

### Proposed Solution

Add a method to grab AWS Managed lists specifically. The user would state which of the domain lists they want and upon deployment, the ID's would be grabbed for them automatically. Since names don't have to be unique for "owned domain lists" I think having a separate method might be the "cleanest" way to do it.

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.1.0

### Environment details (OS name and version, etc.)

Linux

### PR DESCRIPTION
**Pull request by clueleaf**
Currently you have to specify the managed domain list ID manually in order to use it in a DNS firewall rule.
This PR adds a custom resource that finds the ID from a managed domain list name, along with a new `FirewallManagedDomainList` class that creates the custom resource and implements `IFirewallDomainList`.

Closes #25614

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-route53resolver-alpha/test/integ.firewall.ts [hunk 1]
```diff
@@ -1,46 +1,48 @@
...
+
+const blockList = new route53resolver.FirewallDomainList(stack, 'BlockList', {
+  domains: route53resolver.FirewallDomains.fromList(['bad-domain.com', 'bot-domain.net']),
+});
+const overrideList = new route53resolver.FirewallDomainList(stack, 'OverrideList', {
+  domains: route53resolver.FirewallDomains.fromList(['override-domain.com']),
+});
+// const managedMalwareList = new route53resolver.FirewallManagedDomainList(stack, 'ManagedMalwareList', {
+//   managedDomainList: route53resolver.ManagedDomain.MALWARE_DOMAIN_LIST,
+// });
```
Comment (clueleaf) [inline packages/@aws-cdk/aws-route53resolver-alpha/test/integ.firewall.ts:19]: Help wanted:
When I remove the comment out of these lines and try to update the snapshot, custom resource creation fails with the following error:
> Received response status [FAILED] from custom resource. Message returned: Error: connect ECONNREFUSED 127.0.0.1:443 at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) (RequestId: a5f8ab1e-f81c-41af-b887-650d4417bcf7)

I guess there are something wrong with my implementation of custom resource handler so CloudFormation cannot receive the correct callback, but I have no idea.

**Comment by clueleaf** [comment]
Clarification Request
Couldn't update the snapshot because of the reason above.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks (and sorry for the late review) 👍
Looks good!
Just some minor adjustments to be made in my opinion.

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.intrinsics.js.snapshot/asset.b7c985ebfbf370ce93607a33c3851adc0b2b6a530f4c06be487e3c2ec3c06c39.bundle/index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+"use strict";
```
[the remark this answers is not in the document]
Comment (clueleaf) [reply packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.intrinsics.js.snapshot/asset.b7c985ebfbf370ce93607a33c3851adc0b2b6a530f4c06be487e3c2ec3c06c39.bundle/index.js]: It was committed unintentionally. Reverted it.

**Comment by clueleaf** [comment]
@lpizzinidev Thanks for your detailed review! I think your suggestions are all applied.

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Pull request closed without merging**